flowey/flowey_lib_common/src/gh_workflow_id.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use flowey::node::prelude::*;
+
+flowey_request! {
+    pub struct Request {
+        pub github_commit_hash: ReadVar<String>,
+        pub repo_path: ReadVar<PathBuf>,
+        pub gh_workflow_id: WriteVar<String>,
+    }
+}
+
+new_simple_flow_node!(struct Node);
+
+impl SimpleFlowNode for Node {
+    type Request = Request;
+
+    fn imports(_ctx: &mut ImportCtx<'_>) {}
+
+    fn process_request(request: Self::Request, ctx: &mut NodeCtx<'_>) -> anyhow::Result<()> {
+        let Request {
+            repo_path,
+            github_commit_hash,
+            gh_workflow_id,
+        } = request;
+
+        let gh_token = ctx.get_gh_context_var(GhContextVar::GITHUB__TOKEN);
```

Comment:
hmmm, maybe not strictly required for this PR, but do we need some way API like `get_gh_token_with_permissions(impl Iterator<Item = GhPermission>) -> ReadVar<String>`? i.e: in case you need the pipeline to have a particular github permission, outside the context of `emit_gh_step`?